### PR TITLE
IAM | Add Test Cases on IAM API - Part 2

### PR DESCRIPTION
**Pull request by shirady**
### Describe the Problem
Add more test cases to IAM API, this adds if focuses mainly on User and Access Key APIs, mainly on:
- non-exsiting user / non-exsiting access key ID.
- resource deletion before user deletion (access key, IAM policy).
- trying to create an access key above the quota. 

### Explain the Changes
1. Add test cases.

### Issues:
1. List of GAPs:
I plan to add more test cases, mainly in the other APIs (User Inline Policy and Tags APIs).

#### Unit test:
Please run: `npx jest test_string_utils.test.js`

##### Containerized
1. Use the guide "Run Tests From coretest Locally" ([link](https://github.com/noobaa/noobaa-core/blob/master/docs/dev_guide/run_coretest_locally.md)) for running the core tests.
2. Run first tab: `docker run -p 5432:5432 -e POSTGRESQL_ADMIN_PASSWORD=noobaa  quay.io/sclorg/postgresql-15-c9s`
5. Run second tab: `NOOBAA_LOG_LEVEL=all ./node_modules/.bin/mocha src/test/integration_tests/api/iam/test_iam_basic_integration.js`

##### NC:
1. `sudo NC_CORETEST=true ./node_modules/.bin/mocha src/test/integration_tests/api/iam/test_iam_basic_integration.js`
2. `sudo npx jest test_accountspace_fs.test.js` (unit tests)

- [ ] Doc added/updated
- [x] Tests added

## Summary by CodeRabbit

* **Tests**
  * Expanded IAM integration coverage: user lifecycle, path handling, ListUsers path-prefix and requester behaviors, access key creation/update/delete checks, inline policy scenarios, and several negative/delete-edge-case tests.
  * Added unit test for IAM path edge case ('///') and renamed an existing user-update test for clarity.
  * Extended test utilities to support optional IAM paths and inline policy management to enable broader lifecycle/edge-case testing.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/test/unit_tests/util_functions_tests/test_string_utils.test.js [hunk 1]
```diff
@@ -23,6 +23,12 @@ describe('test regex', () => {
             expect(res).toBe(false);
         });
 
+        it('iam path of ///', () => {
+            const valid_path = '///';
+            const res = iam_utils.AWS_IAM_PATH_REGEXP.test(valid_path);
+            expect(res).toBe(true);
+        });
```
[the remark this answers is not in the document]
Comment (shirady) [reply src/test/unit_tests/util_functions_tests/test_string_utils.test.js:30]: I tested this on AWS and it worked, for example:
1. create a user without iam path.
2. update the user and add path of `///`: `aws iam update-user --user-name <some-username> --new-path ///` (worked).

File: src/test/integration_tests/api/iam/test_iam_basic_integration.js [hunk 2]
```diff
@@ -1198,6 +1220,87 @@ mocha.describe('IAM integration tests', async function() {
...
+                        PathPrefix: iam_path
+                    };
+                    const command = new ListUsersCommand(input);
+                    const response = await iam_account.send(command);
+                    _check_status_code_ok(response);
+                    assert.equal(response.Users.length, 2);
+                    for (const user of response.Users) {
+                        assert.equal(user.Path, iam_path);
+                    }
+                });
```
[the remark this answers is not in the document]
Comment (shirady) [reply src/test/integration_tests/api/iam/test_iam_basic_integration.js:1307]: Why is it still open?

**Review by naveenpaul1 — APPROVED** [review]
